autoscaler-example.yaml

```diff
@@ -0,0 +1,73 @@
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: nginx-cluster-proportional-autoscaler-params
+  namespace: default
+data:
+  simple-scalar.scale-map: |-
+    { 
+      "cores_to_replicas_map":
+      [
+        [ 1,1 ],
+        [ 3,3 ],
+        [ 512,5 ],
+        [ 1024,7 ],
+        [ 2048,10 ],
+        [ 4096,15 ],
+        [ 8192,20 ],
+        [ 12288,30 ],
+        [ 16384,40 ],
+        [ 20480,50 ],
+        [ 24576,60 ],
+        [ 28672,70 ],
+        [ 32768,80 ],
+        [ 65535,100 ]
+      ],
+      "nodes_to_replicas_map":
+      [
+        [ 1,1 ],
+        [ 2,2 ]
+      ]
+    }
+---
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: nginx-autoscale-example
+  namespace: default
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        run: nginx-autoscale-example
+    spec:
+      containers:
+      - name: nginx-autoscale-example
+        image: nginx
+        ports:
+        - containerPort: 80
+---
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: cluster-proportional-autoscaler
+  namespace: default
+  labels:
+    version: v1
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        version: v1
+    spec:
+      containers:
+        - image: gcr.io/zihongz-kubernetes-codelab/cluster-proportional-autoscaler:0.6
+          name: autoscaler
+          command:
+            - /cluster-proportional-autoscaler
+            - --namespace=default
+            - --configmap=nginx-cluster-proportional-autoscaler-params
```

Comment:
Good point. I will add a fallback here --- if no namespace is indicated, use the same this pod is on.

For the ConfigMap, we had discussion about this before(#2), and we currently remove the dependency on ConfigMap volumes.  I pasted the previous reasons below:
a) The pod does not start if the ConfigMap volume mount fails (one case was when the ConfigMap was unable to be read by kubelet).
b) Large delay between configmap update and kubelet updating the filesystem.
c) ConfigMap API already provides primary key-level parsing and also avoids a lot of file open/read code.

Though I guess we still need to pass the configmap name even if it is mounted. And I think both ways should work with dynamic updates.